scripts/combine-crossreg.R

```diff
@@ -0,0 +1,237 @@
+         id != results_sponsor_linked_trn # or a self-reference
+  ) |>
+  mutate(is_match_results_sponsor_protocol_id = TRUE) |>
+  select(trn1 = id, trn2 = results_sponsor_linked_trn, is_match_results_sponsor_protocol_id) |>
+  distinct(trn1, trn2, .keep_all = TRUE)
+
+# Integrate new results matches
+trn_trn_results_tidy <- trn_trn_protocols_tidy |>
+  mutate(is_match_results_sponsor_protocol_id = NA) |>
+  rows_upsert(trn_trn_results, by = c("trn1", "trn2"))
+
+
+####################################################################################################################
+# Now add matches made from the title matching algorithm. Will follow similar logic to steps above
+
+# Modify structure of title match table to make it easier to integrate with larger match table
+trn_trn_titles <- title_matches |>
+  filter(id != euctr_id) |>   # filter out self-references
+  select(trn1 = id, trn2 = euctr_id, is_title_matched = has_crossreg_eudract_tm)
+
+# Integrate title matched TRNs
+trn_trn_titles_tidy <- trn_trn_results_tidy |>
+  mutate(is_title_matched = NA) |>
+  rows_upsert(trn_trn_titles, by = c("trn1", "trn2"))
+
+####################################################################################################################
+## Finally add matches from TRNs found in publications
+
+# Unnest TRNs in pubs table
+trn_trn_pubs_long <- publications |>
+  rename(trn1 = primary_IV_id) |>
+  mutate(trn_si_long = strsplit(trns_si, ";"),
+         trn_abs_long = strsplit(trns_abs, ";"),
+         trn_ft_long = strsplit(trns_ft, ";")) |> # Split trns_si by semicolon to get individual trial IDs
+  unnest_longer(trn_si_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_si_long != trn1 | is.na(trn_si_long)) |> # filter out self-references
+  unnest_longer(trn_abs_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_abs_long != trn1 | is.na(trn_abs_long)) |> # filter out self-references
+  unnest_longer(trn_ft_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_ft_long != trn1 | is.na(trn_ft_long)) |> # filter out self-references
+  select(trn1, contains("long"))
+
+# Identify unique TRN pairings from pubs fields, fill out boolean information to identify pub source of match
+trn_trn_pubs <- trn_trn_pubs_long |>
+  rowwise() |>
+  mutate(trn2 = list(unique(c(trn_si_long, trn_abs_long, trn_ft_long)) |> na.omit())) |>
+  unnest_longer(trn2) |>
+  group_by(trn1, trn2) |>
+  summarise(trn2_in_pub_si = if_else(any(trn2 == trn_si_long), TRUE, FALSE),
+            trn2_in_pub_abs = if_else(any(trn2 == trn_abs_long), TRUE, FALSE),
+            trn2_in_pub_ft = if_else(any(trn2 == trn_ft_long), TRUE, FALSE)) |>
+  rowwise() |>
+  mutate(trn1inreg2 = trn1 %in% trn_registries$trn2[trn_registries$trn1 == trn2],
+         trn2inreg1 = trn2 %in% trn_registries$trn2[trn_registries$trn1 == trn1]) |>
+  select(trn1, trn2, trn1inreg2, trn2inreg1, contains("pub")) |>
+  ungroup()
```

Comment:
Here for example I don't follow why `trn1inreg2 and `trn2inreg1` are being filled when adding information from publications

Reply:
@lucfernandez: please address this and @delwen's https://github.com/maia-sh/intovalue-data/pull/44#discussion_r1653306756.

Reply:
My logic was that since publications do not contain any info about trn1inreg2 or trn2inreg1, I should add it in manually here. If we find TRN pairs from publication that aren't anywhere else, removing this step would mean that these pairs have NA for trn1inreg2 or trn2inreg1, something I don't think we want.

However, depending on your thoughts, I can put this step at the end, in the same block where I call which_registry() for all pairs.

Reply:
@lucfernandez so where there is a TRN pair from publication that isn't anywhere else, `trn1inreg2` and `trn2inreg1` would be FALSE?

Conceptually, I would find it clearer to add this to the same block where you call which_registry() for all pairs.

Reply:
This is what I envisioned. `trn1inreg2` and `trn2inreg1` should be FALSE for pairs only found through publications, and all other pairs not linked through registries. 

However, going through my most recent trn_trn table, I see that many rows (anything found through only title matching) have these two fields as NA. I'll add it at the end, with the which_registry() call, and that way all rows will get assigned a value here.